### PR TITLE
Storage Node Peer

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: storagenode/peer.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+// Config is all the configuration parameters for a Storage Node
+type Config struct {
+	Identity identity.Config
+
+	// TODO: switch to using server.Config when Identity has been removed from it
+	PublicAddress string `help:"public address to listen on" default:":7777"`
+	Kademlia      kademlia.Config
+	Piecestore    psserver.Config
```
Comment (mniewrzal) [inline storagenode/peer.go]: For sure `Piecestore`? currently we are using `Storage`. Or my understanding wrong and this won't be used with cmd later.

File: storagenode/peer.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+	var err error
+
+	{ // setup listener and server
+		peer.Public.Listener, err = net.Listen("tcp", config.PublicAddress)
+		if err != nil {
+			return nil, errs.Combine(err, peer.Close())
+		}
+
+		publicConfig := provider.ServerConfig{Address: peer.Public.Listener.Addr().String()}
```
Comment (mniewrzal) [inline storagenode/peer.go]: Why using translation than direct `server.Config`?

Comment (egonelbre) [reply storagenode/peer.go]: Because I forgot to change it. Will fix.

File: storagenode/peer.go (hunk 1 above)
Comment (egonelbre) [reply storagenode/peer.go]: Yeah, storage makes more sense

Comment (egonelbre) [reply storagenode/peer.go]: Fixed

File: storagenode/peer.go (hunk 2 above)
Comment (egonelbre) [reply storagenode/peer.go]: fixed

**Pull request merged**